src/tests/ftest/deployment/critical_integration.py

```diff
@@ -0,0 +1,108 @@
+    """Test Class Description: Verify the basic integration of
+                               the server nodes with available
+                               framework and amongst themselves.
+    :avocado: recursive
+    """
+
+    #TO-DO
+    #Provision all daos nodes using provisioning tool provided by HPCM
+
+    def test_passwdlessssh_versioncheck(self):
+        # pylint: disable=protected-access
+        """
+        Test Description: Verify passwordless ssh amongst the server
+                          server nodes available and verify all server
+                          and client nodes have same daos versions.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=criticalintegration,passwdlessssh_versioncheck
+        """
+
+        check_remote_root_access = self.params.get("check_remote_root_access", "/run/*")
+        daos_server_version_list = []
+        dmg_version_list = []
+        failed_nodes = []
+        for host in self.hostlist_servers:
+            daos_server_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'daos_server version'".format(host))
+            remote_root_access = ("ssh -oNumberOfPasswordPrompts=0 root@{}"
+                                  " 'echo hello'".format(host))
+            command_for_inter_node = ("clush --nostdin -S -w {}"
+                                      " 'echo hello'".format(','.join(self.hostlist_servers)))
+            try:
+                out = run_command(daos_server_cmd)
+                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                if check_remote_root_access:
+                    run_command(remote_root_access)
+                IorTestBase._execute_command(self, command_for_inter_node, hosts=[host])
+            except (DaosTestError, CommandFailure) as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("Ssh check failed on following nodes.\n {}".format(failed_nodes))
+
+        for host in self.hostlist_clients:
+            dmg_version_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'dmg version -i'".format(host))
+
+            try:
+                out = run_command(dmg_version_cmd)
+                dmg_version_list.append(out.stdout.split(b' ')[2])
+            except DaosTestError as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("SSH check for client nodes failed.\n {}".format(failed_nodes))
+
+        result_daos_server = (daos_server_version_list.count(daos_server_version_list[0])
+                              == len(daos_server_version_list))
+        result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
+        result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
```

Comment:
Just FYI, you could also use sets, which contain unique elements:
```suggestion
        result_daos_server = len(set(daos_server_version_list)) == 1
        result_dmg = len(set(dmg_version_list)) == 1
        result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
```

And if the version happened to be in the same format:
```
        same_version = len(set(daos_server_version_list + dmg_version_list)) == 1
```

Reply:
If you don't mind I would like to keep this version check bit as is.